task/jefe/src/main.rs

```diff
@@ -232,6 +232,48 @@ impl idl::InOrderJefeImpl for ServerImpl<'_> {
                 self.dump_areas = dump::initialize_dump_areas();
                 Ok(())
             }
+
+            fn dump_task(
+                &mut self,
+                _msg: &userlib::RecvMessage,
+                task_index: u32,
+            ) -> Result<u8, RequestError<DumpAgentError>> {
+                // `dump::dump_task` doesn't check the task index, because it's
+                // normally called by a trusted source; we'll do it ourself.
+                if task_index == 0 {
+                    // Can't dump the supervisor
+                    return Err(DumpAgentError::NotSupported.into());
+                } else if task_index as usize > self.task_states.len() {
+                    // Can't dump a non-existent task
+                    return Err(DumpAgentError::BadOffset.into());
+                }
+                dump::dump_task(self.dump_areas, task_index as usize)
+                    .map_err(|e| e.into())
+            }
+            fn dump_task_region(
```

Comment:
(would like blank lines between functions, ideally)

Reply:
Me too, not sure where it went!

Fixed in d359897ef132ff95caf40d5ba659feb7fec4459e